### PR TITLE
Export useful types for end-users

### PR DESCRIPTION
**Pull request by jaulz**
As discussed in https://github.com/sindresorhus/got/issues/876 this PR exports most of the types in the index file, e.g. instead of:
```
import { CancelableRequest, Response } from 'got/dist/utils/types'
import { Got, GotStream } from 'got/dist/create'
import { ProxyStream } from 'got/dist/as-stream'
```

it would be:
```
import { Got, GotStream, ProxyStream, CancelableRequest, Response } from 'got'
```

#### Checklist

- [X] I have read the documentation.
- [X] I have included a pull request description of my changes.
- [ ] I have included some tests.
- [ ] If it's a new feature, I have included documentation updates.

### CODE REVIEW

File: source/index.ts [hunk 1]
```diff
@@ -64,3 +64,43 @@ export default got;
...
+export * from './utils/types';
+export {
+	Got,
+	GotStream,
+	ReturnResponse,
+	ReturnStream,
+	GotReturn
+} from './create';
+export {
+	ProxyStream
```
Comment (sindresorhus) [inline source/index.ts]: Not sure `ProxyStream` is a good name to expose.

File: source/index.ts [hunk 2]
```diff
@@ -64,3 +64,43 @@ export default got;
...
+	BeforeErrorHook,
+	AfterResponseHook,
+	HookType,
+	Hooks,
+	HookEvent
+} from './known-hook-events';
+export {
+	GetMethodRedirectCodes,
+	AllMethodRedirectCodes,
+	RequestAsEventEmitter
```
Comment (sindresorhus) [inline source/index.ts]: Do you really need these?

File: source/index.ts (hunk 1 above)
Comment (sindresorhus) [inline source/index.ts]: I think this needs a clearer name.

Comment (jaulz) [reply source/index.ts]: Shall we rename it to `ResponseStream`?

Comment (szmarczak) [reply source/index.ts]: Sounds good to me.

Comment (jaulz) [reply source/index.ts]: Better now?

Comment (sindresorhus) [reply source/index.ts]: Yup

**Comment by jaulz** [comment]
Thanks!

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/got/releases/tag/v10.0.0-alpha.2

**Pull request merged**